webui/react/src/pages/Settings/SettingsAccount.tsx

```diff
@@ -66,39 +99,184 @@ const SettingsAccount: React.FC = () => {
-        <Input maxLength={32} placeholder="Add username" />
-      </InlineForm>
-      <Divider />
-      <InlineForm<string>
-        initialValue={currentUser?.displayName ?? ''}
-        label="Display Name"
-        testId="displayname"
-        onSubmit={handleSaveDisplayName}>
-        <Input maxLength={32} placeholder="Add display name" style={{ widows: '80%' }} />
-      </InlineForm>
-      {info.userManagementEnabled && (
-        <>
-          <Divider />
-          <div className={css.row}>
-            <label>Password</label>
-            <Button onClick={PasswordChangeModal.open}>{CHANGE_PASSWORD_TEXT}</Button>
-          </div>
-          <PasswordChangeModal.Component />
-        </>
-      )}
-    </div>
+    <Drawer open={show} placement="left" title="Settings" onClose={onClose}>
+      <Section title="Profile">
+        <div className={css.section}>
+          <InlineForm<string>
+            initialValue={currentUser?.username ?? ''}
+            label="Username"
+            required
+            rules={[{ message: 'Please input your username', required: true }]}
+            testId="username"
+            onSubmit={handleSaveUsername}>
+            <Input maxLength={32} placeholder="Add username" />
+          </InlineForm>
+          <InlineForm<string>
+            initialValue={currentUser?.displayName ?? ''}
+            label="Display Name"
+            testId="displayname"
+            onSubmit={handleSaveDisplayName}>
+            <Input maxLength={32} placeholder="Add display name" />
+          </InlineForm>
+          {info.userManagementEnabled && (
+            <>
+              <Columns>
+                <Column>
+                  <label>Password</label>
+                </Column>
+                <Button onClick={PasswordChangeModal.open}>{CHANGE_PASSWORD_TEXT}</Button>
+              </Columns>
+              <PasswordChangeModal.Component />
+            </>
+          )}
+        </div>
+        <Divider />
+      </Section>
+      <Section title="Preferences">
+        <div className={css.section}>
+          <InlineForm<Mode>
+            initialValue={currentThemeOption.className}
+            label="Theme Mode"
+            valueFormatter={(value: Mode) => ThemeOptions[value].displayName}
+            onSubmit={(v) => {
+              setMode(v);
+            }}>
+            <Select searchable={false}>
+              <Option key={ThemeOptions.dark.className} value={ThemeOptions.dark.className}>
+                {ThemeOptions.dark.displayName}
+              </Option>
+              <Option key={ThemeOptions.light.className} value={ThemeOptions.light.className}>
+                {ThemeOptions.light.displayName}
+              </Option>
+              <Option key={ThemeOptions.system.className} value={ThemeOptions.system.className}>
+                {ThemeOptions.system.displayName}
+              </Option>
+            </Select>
+          </InlineForm>
+          {Loadable.match(
```

Comment:
I recommend using `Loadable.all` and just handling loading once for the whole drawer.